sound/soc/sof/control.c

```diff
@@ -416,6 +417,14 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 	if (copy_from_user(&header, tlvd, sizeof(const struct snd_ctl_tlv)))
 		return -EFAULT;
 
+	/* Check that length does not exceed the maximum defined for SOF */
+	if (header.length > SND_SOF_CTL_TLV_LENGTH_MAX) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: Bytes size %d exceeds max %d.\n",
+				    header.length, SND_SOF_CTL_TLV_LENGTH_MAX);
+		return -EINVAL;
+	}
+
```

Comment:
can we put this check in a different patch?

Reply:
This check is not that meaningful in a separate patch. If we put this in a separate patch, I'd also use SND_SOF_CTL_TLV_LENGTH_MAX in topology.c:sof_control_load_bytes() where kzalloc() is now done based on value read from user-space.

Reply:
OK, I'll add the same check there as well and separate it into first commit and leave the actual feature fix related code to second commit.

Reply:
Done!